ADD: Overwrite Option

nitincodery opened the pull request:
Check if bb.edn exists, overwrite it or use existing.

nitincodery committed:
ADD: Overwrite Option

Check if bb.edn exists, overwrite it or use existing.

nitincodery committed:
Merge pull request #1 from nitincodery/todo-overwrite-option

ADD: Overwrite Option

nitincodery committed:
FIX: clojure.string --> str

Fixed using var `str` instead of whole namespace.

safehammad commented:
Hey @nitincodery. Thanks so much for contributing! Looking at the PR now :smile:...

safehammad commented on basheetka.bb:
```diff
@@ -100,17 +100,27 @@
 
 (defn help [_]
-  (println  "Usage: ./basheetka.bb init"))
+  (println  "Usage: ./basheetka.bb init [optional] --force"))
```
Minor thought. Do you think it looks more idiomatic to put the option itself in square brackets? Something like:

`Usage: ./basheetka.bb init [--force]`

One day we can use the [built in facility to show help](https://github.com/babashka/cli#help) but that's probably overkill right now.

nitincodery replied:
I thought the same before, I was also not sure how to represent optional sub-command in usage. 

Similarly, strings like "Initializing bb.edn..." changed into "Initializing bb.edn", etc., it's upto you, how you want them to be, I didn't thought about them much.

So, whatever feels right to you, will be much more idiomatic, because I have no idea about conventions to use for defining such strings.

safehammad replied:
> Similarly, strings like "Initializing bb.edn..." changed into "Initializing bb.edn", etc., it's upto you, how you want them to be, I didn't thought about them much.

I think your removal of "..." makes this better :+1:

> So, whatever feels right to you, will be much more idiomatic, because I have no idea about conventions to use for defining such strings.

Let's go with `Usage: ./basheetka.bb init [--force]` if you don't mind, then at some stage in the future, we'll probably reconsider anyway.

safehammad replied:
The only other thing to consider is updating the tests to cover the overwriting (or not) of `bb.edn`. Probably a good idea seeing an error might erase someone's `bb.edn`. If you look in `basheetka_test.bb` you'll see:

```clojure
(deftest round-trip-test
  ...)
```

This might be the right place to consider.

nitincodery replied:
Updated tests for overwrite case too.

I am totally beginner in writing test cases. I am not sure about wether I wrote them correctly or not, specifically about comparing `bb.edn` contents in each test.

safehammad replied:
Great job with the tests!

At some stage, I can see us pulling out the `round-trip-test` into testing the `bb.edn` creation/overwrite functionality separate to the csv file functionality, but we're good for now.

I committed a bit of a tidy up. Feel free to take a look. I like the use of `with-out-str` ... though I think you can use `{:out :string}` as a convenience. Will take a look at that now.

Otherwise, I'll merge in the pull request shortly :)

safehammad reviewed (commented):
Nice addition, thanks! Just a small comment to consider.

nitincodery commented:
I've completed a feature too and currently working on second todo and the excel formula feature, will add them soon too.

nitincodery committed:
FIX: Option syntax clarity improvement

More idiomatic way to put the option itself in square brackets.

nitincodery committed:
ADD: Tests for overwrite update.

Commit:
Minor edits

Commit:
Capture babashka.process shell output using {:out :string}

Rather than `with-out-string` and {:out *out*}, use the
convenience built into babashka.process. The improvement in
clarity is probably marginal.
